### ISSUE TITLE
using the script for intrachromosomal interactions?

### ISSUE THREAD

**Issue by kajocina** [liz-is/readhic#5]
Hi there!

I was wondering if you could give me a hint of how could I got about rewriting your script to load intrachromosomal interactions? It shouldn't be that complicated, I guess?

I would be happy to share the code and commit here ;-)

Best,
P.

**Comment by liz-is** [comment]
Do you mean inter-chromosomal interactions? :)

I haven't looked at this code very recently, but having just refreshed my memory I think there are a few areas that would need changing to enable import of inter-chromosomal interactions. 

Originally I just put a placeholder switch in the function (i.e., `inter_chr=FALSE`), but now I look at the data structure again I think it might be better to write a separate version of the `read_rao_huntley_data` function in order to be able to specify each pair of chromosomes for which you want to read in the data. (And then a wrapper function to read in all data?)

The inter-chromosomal data is a separate download, and the folder structure is slightly different. The path looks like:

`CH12-LX_interchromosomal/1mb_resolution_interchromosomal/chr1_chr2/MAPQGE30/`

which contains one file with interactions between the chromosomes and then three norm files for each chromosome. Right now you could specify `chr = "chr1_chr2"` and the import function would find the right directory. But it would probably be better to be able to specify a vector of chromosomes and find all interactions between all pairs, or to be able to specify two vectors of "first chr" and "second chr". 

> chr1_1mb.KRnorm
> chr1_1mb.SQRTVCnorm
> chr1_1mb.VCnorm
> chr1_2_1mb.RAWobserved
> chr2_1mb.KRnorm
> chr2_1mb.SQRTVCnorm
> chr2_1mb.VCnorm

Each of these files has the same structure as the corresponding files in the directories for the intrachromosomal interactions (i.e., one column per file for the norm vectors, three columns of bin start 1, bin start 2 and observed interactions for the interactions file). So reading in these files is the same. The difference would be that you'd need to create a "bin_df" for each chromosome, rather than just one, to link the bin number with the bin start position. This would probably require writing another version of the `read_data_from_dir` function. 

The GInteractions object is created from a "regions" GRanges, which contains the information for the bin locations, and the interactions between these bins are specified using indices into that regions object. When working with a single chromosome, you can just use the bin number as the index. However, with two chromosomes, the regions object would contain bins from one and then the other, so you'd need to add an offset to the bin number for the second chromosome in order to get the right index. It should be possible to make this work for inter or intra chromosomal interactions with the same function. 

Let me know if you want to give this a go, I'd be happy to have a PR! :)

**Comment by kajocina** [comment]
Thanks for your explanation, this is almost like taking me by hand through this ;-) I just started planning and writing, we'll see what comes out. I will do a commit once it seems to work so then maybe you could merge code and check for any mistakes that might be more obvious to you.

One thing though, inside the interchrosomal interaction folders there are additional normalization vectors included such as "chr2_50kb.INTERKRnorm". Are those important for normalization as well? Or should we just use separately vectors as for intrachromosomal data?

**Comment by liz-is** [comment]
Great, I'll be happy to look over the code when you commit!

I only looked at the data for CH12, not the human data, and I didn't see these extra files there. Maybe there is information about what they are in the README for the data on GEO?

**Comment by kajocina** [comment]
I just checked the supplement of the paper and there they write that the interchromosomal normalization could be only calculated down to 25kb/50kb resolution because the process was a memory hog. I think that those would be the preferred normalization vectors as they exclude intrachromosomal reads although I am not a HiC specialist.

A comparison on the supplement page S19 shows that using interchromosomal KR normalization highlights contact regions which are not standing-out otherwise (with no normalization or just KR norm.)

**Comment by liz-is** [comment]
Yep, I agree that it sounds like those should be used where available. I guess they didn't calculate them for the mouse dataset.

**Comment by kajocina** [comment]
EDIT: found out the answer myself ;-)

**Comment by kajocina** [comment]
I forked and implemented the interchromosomal data handling. It only works by calling a vector with two chromosomes so a user interested in all interactions could just use some sort of iteration over variables. I could also add it later.. Now the new code needs some more testing ;-) I'll tell you if it's looking fine.
